app/controllers/time.ts

```diff
@@ -98,19 +98,20 @@ export default function (controller: botkit.Controller) {
   // Parse a textual punch and produce a new Punch object
   function parse(bot: botkit.Bot, message: Message, mode: string, organization: Organization) {
     mode = mode.toLowerCase();
+    const channelName = message.channel || message.channel_obj.name;
```

Comment:
As I said in my last comment, `message.channel` is just the channel ID on Slack. If you want the channel name (which you do here), you need to use `message.channel_obj.name`, or send the whole `channel_obj` object to get access to the name and ID.